llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1127,19 +1129,11 @@ class APFloat : public APFloatBase {
   /// \param Semantics - type float semantics
   static APFloat getAllOnesValue(const fltSemantics &Semantics);
 
-  /// Returns true if the given semantics supports either NaN or Infinity.
+  /// Returns true if the given semantics supports NaN or Infinity.
   ///
   /// \param Sem - type float semantics
   static bool hasNanOrInf(const fltSemantics &Sem) {
-    switch (SemanticsToEnum(Sem)) {
-    default:
-      return true;
-    // Below Semantics do not support {NaN or Inf}
-    case APFloat::S_Float6E3M2FN:
-    case APFloat::S_Float6E2M3FN:
-    case APFloat::S_Float4E2M1FN:
-      return false;
-    }
```

Comment:
I agree that we do not need to switch-case on types here.

Would returning "sem.nonFiniteBehavior != fltNonfiniteBehavior::FiniteOnly" from here not work?

Reply:
Or maybe even remove it entirely? Users can call `semanticsHasNan` and `semanticsHasInf` instead. The only usage that I see in MLIR is in `APFloatTest.cpp`.

Do you know if the test is actually correct? (I think you wrote that code initially.)
```c++
if (APFloat::hasNanOrInf(Semantics)) {
  EXPECT_FALSE(APFloat::getInf(Semantics, false).isSmallestNormalized());
  EXPECT_FALSE(APFloat::getInf(Semantics, true).isSmallestNormalized());

  EXPECT_FALSE(APFloat::getQNaN(Semantics).isSmallestNormalized());
  EXPECT_FALSE(APFloat::getSNaN(Semantics).isSmallestNormalized());
}
```

Reply:
I believe, it is meant for "not FiniteOnly".

Renaming the API as hasFiniteOnly and then inverting the conditions in APFloatTest.cpp would be cleaner.
What do you think?

Reply:
I moved the function to `APFloatBase` next to `semanticsHasZero` etc.